### ISSUE TITLE
bug(linter): destructuring object with rest property triggers false positive for no-unused-vars

### ISSUE THREAD

**Issue by johnrazeur** [oxc-project/oxc#5258]
When destructuring an object with a rest property, it triggers a false positive when a property is not used:

```tsx
const { id, ...rest } = test
```

```
eslint(no-unused-vars): Variable 'id' is declared but never used.
```

**Comment by johnrazeur** [comment]
Forget to add `ignoreRestSiblings` option  🤦
